src/components/memos/MemoForm.tsx

```diff
@@ -0,0 +1,62 @@
+import React, { useState } from "react";
+
+/** MemoFormコンポーネントのプロパティの型定義 */
+interface MemoFormProps {
+  /**
+   * フォーム送信時に呼び出される関数
+   * @param content 送信されたメモの内容
+   */
+  onSubmit: (content: string) => void;
+}
+
+/**
+ * メモを入力するためのフォームコンポーネント
+ * @param props - コンポーネントのプロパティ
+ * @param props.onSubmit - フォーム送信時に呼び出される関数
+ */
+export const MemoForm: React.FC<MemoFormProps> = ({ onSubmit }) => {
```

Comment:
[MAY] 最近はアロー関数を使ったこっちの書き方は主流ではないですね。パフォーマンスや、IDEでの使いやすさから、以下のような書き方を推奨します。

```suggestion
export default function MemoForm({ onSubmit }: MemoFormProps): React.ReactElement {
```